app/src/main/java/com/github/sdpcoachme/repositories/DefaultRepositoryAccess.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.sdpcoachme.repositories
+
+import com.github.sdpcoachme.database.LineDB
+import com.github.sdpcoachme.database.RequestPoolDB
+import com.github.sdpcoachme.network.DataFormat
+import com.github.sdpcoachme.network.RequestPoolAPI
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import retrofit2.Response
+import java.io.IOException
+import javax.inject.Inject
+import kotlin.random.Random
+
+class DefaultRepositoryAccess @Inject constructor(
+        private val db : RequestPoolDB,
+        private val api : RequestPoolAPI
+) : RepositoryAccess{
+
+    override suspend fun getContentDB(): String = withContext(Dispatchers.IO) {
+        val dbTable : List<LineDB> = db.getAll()
+        var str = ""
+        for (line in dbTable){
+            str = str + line.activity + "\n"
+        }
+        str
+    }
+
+    override suspend fun getRandomDB(): String = withContext(Dispatchers.IO) {
+        val str: String
+        val entriesDB = db.getAll()
+        str = if (entriesDB.isNotEmpty()){
+            val rand = Random.nextInt(entriesDB.size)
+            val randElem = entriesDB[rand]
+            "No internet connection : seeing cached data : " + randElem.activity
+        }else{
+            "No internet connection : no cached data"
+        }
+        str
+    }
+
+    override fun deleteContentDB() {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
It might be a good idea to expose all the methods in the same format, either make them all launch a coroutine, or annotate them all with "suspend" keyword to enforce that they are called from within a coroutine.

Reply:
Another idea could be to use Futures instead and make every function return a Future<SomeType>